care/facility/api/serializers/patient.py

```diff
@@ -539,6 +561,17 @@ def create(self, validated_data):
             # If the user is not a doctor then the user type is the same as the user type
             validated_data["user_type"] = user_type
 
+        if validated_data.get("reply_to"):
+            reply_to_note = validated_data["reply_to"]
+            if reply_to_note.thread != validated_data["thread"]:
+                raise serializers.ValidationError(
+                    "Reply to note should be in the same thread"
+                )
+            if reply_to_note.consultation != validated_data.get("consultation"):
```

Comment:
consultation is not a required field, need better validations

Reply:
Could you please suggest what additional validations can be implemented?